infra/syntax/Json.cpp

```diff
@@ -631,6 +650,28 @@ namespace infra
             return JsonToken::Error();
     }
 
+    JsonToken::Token JsonTokenizer::TryCreateFloatToken(uint64_t integer, bool sign)
+    {
+        if (parseIndex != objectString.size() && objectString[parseIndex] == '-')
+            return JsonToken::Error();
+
+        auto tokenStart = parseIndex;
+
+        while (parseIndex != objectString.size() && std::isdigit(static_cast<unsigned char>(objectString[parseIndex])))
+            ++parseIndex;
+
+        infra::BoundedConstString fractional = objectString.substr(tokenStart, parseIndex - tokenStart);
+        fractional.shrink(nanoValueWidth);
+
+        auto fractionalValue = ToInt<uint32_t>(fractional, false);
+
+        if (fractional.size() < 9)
+            for (std::size_t count = fractional.size(); count < nanoValueWidth; ++count)
+                fractionalValue = fractionalValue * 10;
+
+        return JsonFloat(static_cast<uint32_t>(integer), fractionalValue, sign);
```

Comment:
Instead of casting the integer part to 32 bits, I would rather see JsonFloat accept 64 bits